relay-ourlogs/src/ourlog.rs

```diff
@@ -70,9 +72,152 @@ pub fn otel_to_sentry_log(otel_log: OtelLog) -> OurLog {
+            let span_id = if event_trace_id.is_some() {
+                breadcrumb
+                    .data
+                    .value()
+                    .and_then(|data| data["__span"].value())
+                    .and_then(|span| match span {
+                        Value::String(s) => Some(Annotated::new(SpanId(s.clone()))),
+                        _ => None,
+                    })
+                    .unwrap_or_else(Annotated::empty)
+            } else {
+                Annotated::empty()
+            };
+
+            // Convert breadcrumb data fields to primitive attributes
+            if let Some(data) = breadcrumb.data.value() {
+                for (key, value) in data.iter() {
+                    if let Some(value) = value.value() {
+                        let attribute = match value {
+                            Value::String(s) => Some(AttributeValue::StringValue(s.clone())),
+                            Value::Bool(b) => Some(AttributeValue::BoolValue(*b)),
+                            Value::I64(i) => Some(AttributeValue::IntValue(*i)),
+                            Value::F64(f) => Some(AttributeValue::DoubleValue(*f)),
+                            _ => None, // Complex types will be supported once consumers are updated to ingest them.
+                        };
+
+                        if let Some(attr) = attribute {
+                            attribute_data.insert(key.clone(), Annotated::new(attr));
+                        }
+                    }
+                }
+            }
+
+            let (body, level) = match breadcrumb.ty.value().map(|ty| ty.as_str()) {
+                Some("http") => format_http_breadcrumb(breadcrumb)?,
+                Some(_) | None => format_default_breadcrumb(breadcrumb)?,
+            };
+
+            Some(OurLog {
+                timestamp_nanos: Annotated::new(timestamp_nanos),
+                observed_timestamp_nanos: Annotated::new(timestamp_nanos),
+                trace_id: event_trace_id
+                    .clone()
+                    .map(Annotated::new)
+                    .unwrap_or_else(Annotated::empty),
+                span_id,
+                trace_flags: Annotated::new(0),
+                severity_text: level,
+                severity_number: Annotated::empty(),
+                body,
+                attributes: Annotated::new(attribute_data),
+                ..Default::default()
+            })
+        })
+        .collect()
+}
+
+fn format_http_breadcrumb(
+    breadcrumb: &Breadcrumb,
+) -> Option<(Annotated<String>, Annotated<String>)> {
+    let data = breadcrumb.data.value().cloned().unwrap_or_default();
+
+    match (
+        data.get("method").and_then(|v| v.value()),
+        data.get("status_code").and_then(|v| v.value()),
+        data.get("url").and_then(|v| v.value()),
+    ) {
+        (Some(Value::String(method)), Some(Value::I64(status)), Some(Value::String(url))) => {
```

Comment:
Complicated way to do this. Either a `if let (Some(_), Some(_), Some(_)) else {}` or the proper way:

```rs
        let method = data.get("method").and_then(|v| v.value())?;
        let status_code = data.get("status_code").and_then(|v| v.value())?;
        let url = data.get("url").and_then(|v| v.value())?;
```